app/database/operator/server_data_operator/transformers/post.ts

```diff
@@ -151,3 +153,36 @@ export const transformPostsInChannelRecord = ({action, database, value}: Transfo
         fieldsMapper,
     }) as Promise<PostsInChannelModel>;
 };
+
+/**
+ * transformPostRecords: Prepares records of the SERVER database 'ScheduledPosts' table for update or create actions.
+ */
+export const transformSchedulePostsRecord = ({action, database, value}: TransformerArgs): Promise<ScheduledPostModel> => {
+    const emptyFileInfo: FileInfo[] = [];
+    const emptyPostMetadata: PostMetadata = {};
+    const raw = value.raw as ScheduledPost;
+
+    const fieldsMapper = (scheduledPost: ScheduledPostModel) => {
+        scheduledPost._raw.id = scheduledPost.id;
+        scheduledPost.rootId = raw?.root_id ?? '';
+        scheduledPost.message = raw?.message ?? '';
+        scheduledPost.channelId = raw?.channel_id ?? '';
+        scheduledPost.files = raw?.files ?? emptyFileInfo;
+        scheduledPost.metadata = raw?.metadata ?? emptyPostMetadata;
```

Comment:
In the table schema you said that `metadata` could be optional, but you are always assigning a value. Why is that?

Reply:
Yup, this is optional, so we can assign the default value of `undefined` to it. Nice catch. I have taken the reference from the draft transformer and it does the same thing.

Reply:
Optional in the db means null not undefined

Reply:
Thanks, @enahum, got it.